### ISSUE TITLE
match_like_matches_macro does not lint on T-F-T-style matches

### ISSUE THREAD

**Issue by CommanderStorm** [rust-lang/rust-clippy#17616]
### Summary

The middle detection of match_like_matches_macro could be more eager.

Below code is not linted because the lint currently requires the first and middle arms to return the same boolean value.
Here, the middle arm returns false while the first and last arms return true.

We can do better 😉

### Lint Name

match_like_matches_macro

### Reproducer

I tried this code:

```rust
#[warn(match_like_matches_macro)]
fn foo(){
    match match_type {
        MatchType::A(_str1) => true, // this bool != next one
        MatchType::B(_str1, _str2) => false,
        MatchType::C => true,
    }
}
```

I expected to see this suggestion happen:

```rust
#[warn(match_like_matches_macro)]
fn foo(){
    matches!(match_type(MatchType::A(_) | MatchType::C)
}
```

Instead, this happened:
no lint

Noticed in @wasd243's PR:
- https://github.com/rust-lang/rust-clippy/pull/17505#discussion_r3838474503

### Version

```text
rustc 1.98.0 (88d9e12ae 2026-08-18)
binary: rustc
commit-hash: 88d9e12ae178fab0fb5cc050a94da85685d449ea
commit-date: 2026-08-18
host: x86_64-unknown-linux-gnu
release: 1.98.0
LLVM version: 22.1.8
```
